### PR TITLE
Add ball tracking telemetry and more useful control commands

### PR DESCRIPTION
**Pull request by WasabiFan**
Also fixed some misc bugs.

### CODE REVIEW

File: CompetitionTemplate/src/competition/subsystems/collector/CollectorSubsystem.java [hunk 1]
```diff
@@ -19,16 +24,31 @@
...
+        intakePower = propMan.createPersistentProperty("Collector intake power", 1.0);
+        ejectPower = propMan.createPersistentProperty("Collector eject power", -1.0);
     }
     
-    public void setIntakePower(double power) {
+    public void setCollectorPower(double power) {
         this.collectorMotorLeft.set(power);
     }
     
+    public void startEject() {
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/collector/CollectorSubsystem.java]: remove start from startEject() and startIntake() - just Intake and Eject are fine.

File: CompetitionTemplate/src/competition/subsystems/vision/VisionSubsystem.java [hunk 2]
```diff
@@ -35,11 +32,29 @@ public VisionSubsystem(JetsonServerManager jetsonServer, WPIFactory factory, XPr
...
+    }
+    
     public boolean isConnectionHealthy() {
         return jetsonServer.isConnectionHealthy();
     }
-    
-    public void updateMonitorLogging() {
-        this.monitor.update();
+
+    public boolean trackingAnyBalls() {
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/vision/VisionSubsystem.java]: to match the other boolean getters, call this "isTrackingAnyBalls()"

File: CompetitionTemplate/src/competition/subsystems/vision/commands/CollectForwardBallCommand.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        
+        requires(driveSubsystem);
+        requires(collectorSubsystem);
+
+        powerProperty = propMan.createPersistentProperty("Vision collect speed", 0.4);
+    }
+
+    @Override
+    public void initialize() {
+        BallSpatialInfo targetBall = visionSubsystem.findTargetBall();
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/vision/commands/CollectForwardBallCommand.java]: You should have some sanity check in here for distance

File: CompetitionTemplate/src/competition/subsystems/vision/commands/CollectForwardBallCommand.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        targetDistance = targetBall.distanceInches + 12;
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/vision/commands/CollectForwardBallCommand.java]: targetBall can return null.

File: CompetitionTemplate/src/competition/subsystems/vision/commands/RotateTowardsBallCommand.java [hunk 5]
```diff
@@ -27,6 +29,7 @@ public RotateTowardsBallCommand(VisionSubsystem visionSubsystem, DriveSubsystem
         requires(driveSubsystem);
         
         rotationalPidManager = new PIDManager("Ball rotation", propMan, 1d/20d, 0, 0);
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/vision/commands/RotateTowardsBallCommand.java]: Just put this as 0.05 - that way it'll look the same on the dashboard

File: CompetitionTemplate/src/competition/subsystems/vision/commands/RotateTowardsBallCommand.java [hunk 6]
```diff
@@ -35,25 +38,16 @@ public void initialize() {
...
-            BallSpatialInfo targetBall = null;
-            for(BallSpatialInfo ball : ballInfo) {
-                if(targetBall == null || ball.distanceInches > targetBall.distanceInches) {
-                    targetBall = ball;
-                }
-            }
+            BallSpatialInfo targetBall = visionSubsystem.findTargetBall();
 
-            double newRotationalPower = rotationalPidManager.calculate(0, targetBall.relativeHeading);
+            double newRotationalPower = rotationalPidManager.calculate(cameraCenterHeading.get(), targetBall.relativeHeading);
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/vision/commands/RotateTowardsBallCommand.java]: Is there a potential race condition where targetBall could be null due to the subsystem updating on an independent thread?

**Comment by WasabiFan** [comment]
Updated.

**Pull request merged**